sklearn/utils/_mocking.py

```diff
@@ -196,6 +204,13 @@ def fit(self, X, y, **fit_params):
                         f'Fit parameter {key} has length {_num_samples(value)}'
                         f'; expected {_num_samples(X)}.'
                     )
+        if self.expected_sample_weight:
+            if sample_weight is None:
+                raise AssertionError("Expected sample_weight to be passed")
+            if len(sample_weight) != _num_samples(X):
```

Comment:
It would be best to use `sklearn.utils.validation._check_sample_weight`

Reply:
:+1: